### PR TITLE
Optional handler for unauthorized access

### PR DESCRIPTION
**Pull request by zatziky**
Hi @burczu,

I hope you can accept this PR. Sometimes we don't want to be redirected to `this.notAuthorizedPath`. In this PR I am proposing a custom handler of unauthorized paths while keeping the previous behaviour. 

Let me know how you like it and what I shall so that you would accept it.

Cheers!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -114,4 +114,44 @@ But this is not all. The component provides two methods: `this.rolesMatched` and
 - `this.rolesMatched` finds the intersection of the two arrays and returns `true` if at least one of the user roles is present among the available roles.
 - `this.rolesMatchedExact` checks if the available roles array has exactly the same items as the user roles array.
 
-As you can see in the example above, you can use one of these methods to return the markup of the component or just `null`.
\ No newline at end of file
+As you can see in the example above, you can use one of these methods to return the markup of the component or just `null`.
+
+### Custom handling of unauthorized access
```
Comment (burczu) [inline README.md]: I would suggest to make it a sub-paragraph of the AuthorizedComponent paragraph

File: lib/components/AuthorizedComponent.react.js [hunk 2]
```diff
@@ -25,16 +25,37 @@ class AuthorizedComponent extends React.Component {
...
+     * @param routeRoles roles defending the route requested
+     * @param userRoles roles of the user accessing the component
+     */
+    //noinspection JSUnusedLocalSymbols - the parameters may be used by subclasses
+  handleUnauthorizedRole(routeRoles, userRoles){
+    const {router} = this.context;
+    router.push(this.notAuthorizedPath);
+  }
+
+  resolveUnauthorizedRoleHandler() {
```
Comment (burczu) [inline lib/components/AuthorizedComponent.react.js]: what is this for? I can't see any usage of this method...

**Review by burczu — COMMENTED** [review]
generally I like this PR, but please refer to my comments

File: README.md (hunk 1 above)
Comment (zatziky) [reply README.md]: ok

File: lib/components/AuthorizedComponent.react.js (hunk 2 above)
Comment (zatziky) [reply lib/components/AuthorizedComponent.react.js]: It's not used, thx! Removed.

**Review by burczu — APPROVED** [review]

**Pull request merged**